scene/animation/animation_node_state_machine.cpp

```diff
@@ -1263,9 +1264,15 @@ void AnimationNodeStateMachine::add_node(const StringName &p_name, Ref<Animation
 	ERR_FAIL_COND(p_node.is_null());
 	ERR_FAIL_COND(String(p_name).contains("/"));
 
+	PackedInt32Array node_group_selections;
+	node_group_selections.resize(12);
+	node_group_selections.fill(0);
+
 	State state_new;
 	state_new.node = p_node;
 	state_new.position = p_position;
+	state_new.node_group_selections = node_group_selections;
+	state_new.node_visibility = true;
```

Comment:
```suggestion
	State state_new;
	state_new.node = p_node;
	state_new.position = p_position;
#ifdef TOOLS_ENABLED
	state_new.node_group_selections = node_group_selections;
	state_new.node_visibility = true;

	PackedInt32Array node_group_selections;
	node_group_selections.resize(12);
	node_group_selections.fill(0);
#endif
```

If you want to save these data for each StateMachine, we can allow only the editor to do so by saving them in TOOLS_ENABLED.